docs/proposals/08-shoot-additional-container-runtimes.md

```diff
@@ -0,0 +1,139 @@
+        
+    2. Containerd socket path and containerd configurations are different for each OS. To make sure the configurations above are set in each worker machine, each OS extension would be responsible to configure them during the reconciliation of the 
+       OperatingSystemConfig when it creates the cloud init.
+        1. os-ubuntu - 
+            1. Add the following flags to the kubelet execution command:
+                1. --container-runtime=remote
+                2. --container-runtime-endpoint=unix:///run/containerd/containerd.sock
+            2. Create a containerd configuration file: /etc/containerd/config.toml based on the current default configuration (containerd config default).
+        2. os-coreos - 
+            1. Add the following flags to the kubelet execution command:
+               1. --container-runtime=remote
+               2. --container-runtime-endpoint=unix:///var/run/docker/libcontainerd/docker-containerd.socket
+            2. Create a containerd configuration file: /etc/containerd/config.toml based on the file in the path: __TBD__ and remove the CRI disabled plugin. 
+            3. Override the default containerd.service unit. Create a new unit in: /etc/systemd/system/containerd.service unit that will start the containerd service with a configuration file path: /etc/containerd/config.toml.
+        3. os-suse-jeos - __TBD__     
+        
+    3. Docker-monitor daemon and rotate log should be replaced with equivalent conatinerd services. __TBD__ 
+
+2. Validate workers additional runtime configurations.  __TBD__
+3. Add support for each additional container runtime in the cluster.   
+    1. In order to install each additional available runtime in the cluster we should:
+        1. Install the runtime binaries in each Worker's pool nodes that specified the runtime support.
+        2. Apply the relevant RuntimeClass to the cluster.
+        
+    2. The installation above should be done by a new kind of extension: ContainerRuntime resource. For each container runtime type (Kata-container/gvisor) a dedicate extension controller will be created. 
+        1. A label for each container runtime support will be added to every node that belongs to the worker pool. This should be done similar
+           to the way labels created today for each node, through kubelet execution parameters (_kubelet.flags: --node-labels). When creating the OperatingSystemConfig (original) for the worker each container runtime support should be mapped to a label on the node.
+           For Example:
+           label: container.runtime.kata-containers=true (shoot.spec.cloud.<IAAS>.worker.containerRuntimes.kata-container)
+           label: container.runtime.gvisor=true (shoot.spec.cloud.<IAAS>.worker.containerRuntimes.gvisor)
+        2. During the Shoot reconciliation (Similar steps to the Extensions today) Gardener will create new ContainerRuntime resource if a container runtime exist in at least one worker spec:
```

Comment:
What will happen if I have 2 worker groups, both are using gVisor but with different configurations?
I would suggest to use one resrouce per Shoot how it is done for the `Worker`, or every runtime in every worker group to have its own `ContainerRuntime` resource.

Reply:
Supporting the ability to configure different parameters for each type of runtimeclass is currently a non-goal.

Reply:
We will transform into an object instead of array of strings to allow future improvement.